### PR TITLE
Adding headers to validation

### PR DESCRIPTION
**Pull request by ericaovo**
PR related to [this](https://github.com/joanvila/raml-js-validator/issues/14) issue.

@joanvila apologies in advance, I'm a back-end dev, this is my first JS PR! Any suggestion is more than welcome! :)

### CODE REVIEW

File: lib/endpoint-builder.js [hunk 1]
```diff
@@ -66,6 +66,20 @@ const EndpointBuilder = function() {
         return baseUri;
     };
 
+    this.parseHeaders = (headers) => {
+        if (!headers.length) {
+            return null;
+        }
+
+        var headersMap = {};
```
Comment (joanvila) [inline lib/endpoint-builder.js]: Please use `let` instad of `var` here. Apart from that, I don't see why we need the first if. Wouldn't work just skipping that and the function would return the empty `headersMap` object?

File: lib/endpoint-checker.js [hunk 2]
```diff
@@ -10,12 +10,14 @@ const EndpointChecker = function() {
...
     this.responseSchemaChecker = new ResponseSchemaChecker();
 
-    this.check = (endpointToTest, method, expectedResponses, postData, validateResponse) => {
+    this.check = (endpointToTest, method, apiHeaders, expectedResponses, postData, validateResponse) => {
         let d = Q.defer();
 
         let expectedResponseCodes = getExpectedResponseCodes(expectedResponses);
 
-        request[method](endpointToTest, {form: postData}, (error, response, body) => {
+        var options = {url: endpointToTest, headers: apiHeaders, form: postData};
```
Comment (joanvila) [inline lib/endpoint-checker.js]: Please, use `const` instead of `var` here

File: lib/validator.js [hunk 3]
```diff
@@ -39,16 +39,25 @@ const Validator = function(api, target, validateResponse) {
...
                 this.endpointChecker.check(
                     endpointToTest,
                     methodName,
+                    apiHeaders,
                     expectedResponses,
                     postData,
                     validateResponse
                 ).then(() => {
-                    console.log('[OK][' + methodName + '] ' + endpointToTest);
+                    var headers = '';
```
Comment (joanvila) [inline lib/validator.js]: Use `let` instead of `var` here

**Comment by joanvila** [comment]
Good job @ericaovo I've just added some small comments. Once applied I'm happy to merge it 🙂

**Comment by ericaovo** [comment]
@joanvila I'd be more than happy to fix them, but I can only see the coverage checks from coveralls: are you referring to those?

**Comment by joanvila** [comment]
Sorry @ericaovo I didn't publish the comments

File: lib/endpoint-builder.js (hunk 1 above)
Comment (ericaovo) [reply lib/endpoint-builder.js]: It would, changed.

File: lib/endpoint-checker.js (hunk 2 above)
Comment (ericaovo) [reply lib/endpoint-checker.js]: Done

File: lib/validator.js (hunk 3 above)
Comment (ericaovo) [reply lib/validator.js]: Done

**Comment by ericaovo** [comment]
No worries @joanvila, addressed em all!

**Comment by mwz** [comment]
@joanvila, any plans to get this merged/released soon?

**Comment by joanvila** [comment]
Sorry for the delay @ericaovo and @mwz it's merged now! I'll push a new version to npm now

**Pull request merged**